config: support config schema evolution (#9461)
tarantool/tarantool: changelogs/unreleased/config-drop-config-version.md

## feature/config

> * `config.version` option is dropped, because another schema evolution
>   mechanism is implemented (gh-9452).

Reviewer: ```suggestion
* Removed the `config.version` option because a different schema evolution
  mechanism was implemented (gh-9452).
```

Author: Thanks, updated.
